src/main/java/com/mraof/minestuck/blockentity/machine/PunchDesignixBlockEntity.java

```diff
@@ -114,8 +117,13 @@ private void handleKeyboardClick(ServerPlayer player)
 		{
 			if(AlchemyHelper.hasDecodedItem(heldStack))
 			{
-				ItemStack output;
-				if(AlchemyHelper.isPunchedCard(getCard()))	//|| combination
+				ItemStack output = AlchemyHelper.getDecodedItem(heldStack);
+				if(output.is(MSTags.Items.UNREADABLE) && !AlchemyHelper.isReadableCard(output))
+				{
+					player.sendSystemMessage(Component.translatable("On closer inspection, the code on the back seems to be almost unreadable").withStyle(ChatFormatting.BOLD));
```

Comment:
You're using a string literal instead of a translation key for the translatable message. You should change it to use a translation key which you then specify the text for in `MinestuckEnUsLanguageProvider`.

Reply:
can do